subprojects/launcher/src/main/java/org/gradle/launcher/daemon/server/DefaultIncomingConnectionHandler.java

```diff
@@ -157,7 +157,7 @@ private void handleCommand(Command command, DaemonConnection daemonConnection) {
             try {
                 if (!Arrays.equals(command.getToken(), token)) {
                     throw new BadlyFormedRequestException(String.format("Unexpected authentication token in command %s received from %s, expected: %s, incoming: %s.",
-                        command, connection, Arrays.toString(token), Arrays.toString(token)));
+                        command, connection, Arrays.toString(token), Arrays.toString(command.getToken())));
```

Comment:
hmm, I don't think we should do this. This would allow you to get the token by sending a wrong one first.

Maybe we can include the daemonContext UID to identify which daemon we were talking to?

Reply:
We are already logging the token, don't we? Here in this exception and here: https://github.com/gradle/gradle/pull/15324#discussion_r573573263

Please propose something, I am also fine abandoning this PR.